packages/wdio-utils/tests/envDetector.test.ts

```diff
@@ -46,14 +46,17 @@ describe('sessionEnvironmentDetector', () => {
     it('isW3C', () => {
         const requestedCapabilities = { browserName: '' }
         expect(sessionEnvironmentDetector({ capabilities: {}, requestedCapabilities: {} }).isW3C).toBe(false)
-        expect(sessionEnvironmentDetector({ capabilities: appiumCaps, requestedCapabilities }).isW3C).toBe(true)
-        expect(sessionEnvironmentDetector({ capabilities: experitestAppiumCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: chromeCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: geckoCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: safariCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: safariDockerNbVCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: safariDockerNpNCaps, requestedCapabilities }).isW3C).toBe(true)
         expect(sessionEnvironmentDetector({ capabilities: edgeCaps, requestedCapabilities }).isW3C).toBe(true)
+        /**
+         * Legacy / JWP responses
+         */
+        expect(sessionEnvironmentDetector({ capabilities: appiumCaps, requestedCapabilities }).isW3C).toBe(false)
```

Comment:
@christian-bromann 

The W3C check is used validating the caps before the driver starts and for returned caps from the driver (sessionEnvironmentDetector) The first one CAN properly check if the Appium response is W3C, the second one CAN NOT, at least not what I can find.

Should we have a different `sessionEnvironmentDetector` + `isW3C` check?

Reply:
> The W3C check is used validating the caps before the driver starts

It is indeed a mixture out of received capabilities and requested ones. There is not really a deterministic way to define this, especially when using vendors like Sauce Labs.

What are the advantages of creating two different methods for it? The `sessionEnvironmentDetector` method gets both requested and received capabilities and therefor has everything to determine the environment.

Reply:
Well, there are a few things that we need to be aware of:

- Appium `1.x` is JWP/W3C compatible, but not strict on W3C caps. Meaning you can still send W3C commands to a session which is started with JWP caps
- Appium `2.x` is / should be W3C strict, also on the caps
- Appium `1.x` will be used for a long time by users / cloud vendors so we need to support that
- As fas as I know / form what I've seen is that Appium `1.x` uses caps that are JWP, but also W3C compatible (like `platformName`)
- As far as I've seen we can't rely on the `sessionEnvironmentDetector` to determine correctly if it was a W3C Appium session (from what I've seen the `appium:` prefixes are stripped out in the session response

I'm not sure if we should handle that by a single W3C method that is used for both received capabilities and requested ones, but I can be wrong 😉